### PR TITLE
enhancing compatibility with legacy ORC files

### PR DESCRIPTION
**Pull request by ico01**
## Description
This PR modifies the handling of the hive.orc.use-column-names configuration setting in Presto to allow for improved compatibility with older ORC data that may not contain column names in the file. The new strategy uses ORC file column names by default and falls back to Hive schema column names when ORC names are not present.

## Motivation and Context
The ability for Presto to flexibly handle column names in ORC files is critical for accessing a wide range of data, including legacy datasets that may not contain embedded column names. The original hive.orc.use-column-names configuration setting only allowed access via column names and did not account for files without them. This approach helps mitigate issues with reading old ORC data and ensures that users can access their datasets consistently without manual intervention.

## Impact
The changes to hive.orc.use-column-names represent an enhancement to the existing functionality rather than a breaking change. Users specifying this option will now benefit from an intelligent fallback mechanism. There should be no performance impact as the fallback only occurs when required, and there are no changes to the public API.

## Test Plan
The changes have been tested using both new and old ORC files—both those with embedded column names and those without. The Presto cluster correctly accessed column data using the appropriate naming strategy in each case. 

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Hive Changes
* Improves the `hive.orc.use-column-names` configuration setting to no longer fail on reading ORC files 
  without column names but falls back to using Hive's schema, enhancing compatibility with legacy ORC files.
```

### CODE REVIEW

**Comment by jaystarshot** [comment]
Is it possible to add unit testing? I can see some related tests at  testHiveFileFormats

**Comment by ico01** [comment]
> Is it possible to add unit testing? I can see some related tests at testHiveFileFormats

ok, i just did't know where to test

**Comment by ico01** [comment]
@jaystarshot i add unit test for it

**Review by jaystarshot — DISMISSED** [review]
lgtm, deferring to a orc expert for any concerns

**Comment by tdcmeehan** [comment]
CC: @sdruzkin

**Comment by ico01** [comment]
@tdcmeehan @sdruzkin @mbasmanova It's has been two weeks since last action, what should i do now

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java
Comment (tdcmeehan) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java:1172]:
```suggestion
    private static boolean fileHasColumnNames(List<String> physicalColumnNames, Path path)
```

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java
Comment (tdcmeehan) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java:1178]:
```java
    return physicalColumnNames.isEmpty || !physicalColumnNames.stream().allMatch(physicalColumnName -> DEFAULT_HIVE_COLUMN_NAME_PATTERN.matcher(physicalColumnName).matches())
```

**Review by tdcmeehan — CHANGES_REQUESTED** [review]
Some nits.  Overall, after reading through the ORC spec, this change seems fine.  Pinging @sdruzkin one last time as our ORC expert, otherwise I'm happy to merge once the nits are addressed.

**Comment by ico01** [comment]
thanks,I will fix these

**Comment by steveburnett** [comment]
Suggest revising release note entry following the [Order of changes](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes) in the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines). Perhaps something like this: 

```
== RELEASE NOTES ==

Hive Changes
* Improves the `hive.orc.use-column-names` configuration setting to no longer fail on reading ORC files 
  without column names but falls back to using Hive's schema, enhancing compatibility with legacy ORC files.
```

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java
Comment (ico01) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java]: fixed

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java
Comment (ico01) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java]: fixed

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java
Comment (ico01) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java]: fixed

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java
Comment (ico01) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: fixed

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java
Comment (ico01) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: fixed

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java
Comment (ico01) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: fixed

**Review by tdcmeehan — APPROVED** [review]

**Comment by ico01** [comment]
i found it's very difficult to run all check passed, each time it will have different test not passed. even though i didn't change code

**Comment by tdcmeehan** [comment]
@ico01 the CI environment has become unstable lately.  We are working on making it more stable again.  In the meantime, maintainers will ensure tests pass and will merge.

**Comment by ico01** [comment]
ok

**Pull request merged**
